src/hooks/useProfitAndLoss/useProfitAndLossDetailLines.tsx

```diff
@@ -0,0 +1,573 @@
+  ),
+})
+
+const OpeningBalanceLedgerEntrySourceSchema = Schema.Struct({
+  displayDescription: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('display_description'),
+  ),
+  entityName: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('entity_name'),
+  ),
+  type: Schema.Literal('Opening_Balance_Ledger_Entry_Source'),
+  accountName: pipe(
+    Schema.optional(Schema.String),
+    Schema.fromKey('account_name'),
+  ),
+  memo: Schema.optional(Schema.NullOr(Schema.String)),
+  metadata: Schema.optional(Schema.NullOr(Schema.Unknown)),
+  referenceNumber: pipe(
+    Schema.optional(Schema.NullOr(Schema.String)),
+    Schema.fromKey('reference_number'),
+  ),
+})
+
+const PayoutLedgerEntrySourceSchema = Schema.Struct({
+  displayDescription: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('display_description'),
+  ),
+  entityName: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('entity_name'),
+  ),
+  type: Schema.Literal('Payout_Ledger_Entry_Source'),
+  payoutId: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('payout_id'),
+  ),
+  externalId: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('external_id'),
+  ),
+  paidOutAmount: pipe(
+    Schema.propertySignature(Schema.Number),
+    Schema.fromKey('paid_out_amount'),
+  ),
+  processor: Schema.String,
+  completedAt: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('completed_at'),
+  ),
+  memo: Schema.optional(Schema.NullOr(Schema.String)),
+  metadata: Schema.optional(Schema.NullOr(Schema.Unknown)),
+  referenceNumber: pipe(
+    Schema.optional(Schema.NullOr(Schema.String)),
+    Schema.fromKey('reference_number'),
+  ),
+})
+
+export const LedgerEntrySourceSchema = Schema.Union(
+  TransactionLedgerEntrySourceSchema,
+  InvoiceLedgerEntrySourceSchema,
+  ManualLedgerEntrySourceSchema,
+  InvoicePaymentLedgerEntrySourceSchema,
+  RefundLedgerEntrySourceSchema,
+  RefundPaymentLedgerEntrySourceSchema,
+  OpeningBalanceLedgerEntrySourceSchema,
+  PayoutLedgerEntrySourceSchema,
+)
+
+const AccountTypeSchema = Schema.Struct({
+  value: Schema.String,
+  displayName: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('display_name'),
+  ),
+})
+
+const AccountSubtypeSchema = Schema.Struct({
+  value: Schema.String,
+  displayName: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('display_name'),
+  ),
+})
+
+export const AccountSchema = Schema.Struct({
+  id: Schema.String,
+  name: Schema.String,
+  stableName: pipe(
+    Schema.propertySignature(Schema.String),
+    Schema.fromKey('stable_name'),
+  ),
+  normality: Schema.String,
+  accountType: pipe(
+    Schema.propertySignature(AccountTypeSchema),
+    Schema.fromKey('account_type'),
+  ),
+  accountSubtype: pipe(
+    Schema.propertySignature(AccountSubtypeSchema),
+    Schema.fromKey('account_subtype'),
+  ),
```

Comment:
I think we can reuse the AccountTypeSchema unless we accountType and accountSubtype are going to diverge.
```suggestion
  accountSubtype: pipe(
    Schema.propertySignature(AccountTypeSchema),
    Schema.fromKey('account_subtype'),
  ),
```